libgalois/include/katana/EntityTypeManager.h

```diff
@@ -85,12 +86,16 @@ class EntityTypeManager {
 
   /// adds a new entity type for the atomic type with name \p name
   /// \returns the EntityTypeID for the new type
-  EntityTypeID AddAtomicEntityType(const std::string& name) {
+  Result<EntityTypeID> AddAtomicEntityType(const std::string& name) {
     KATANA_LOG_DEBUG_ASSERT(!HasAtomicType(name));
 
-    // TODO(roshan) limit the new entity type id to the max for 8 bits
-    // and return an error
     EntityTypeID new_entity_type_id = GetNumEntityTypes();
+    if (new_entity_type_id >= kInvalidEntityType) {
```

Comment:
nit: the type enforces that it cannot be greater

Reply:
Yea. But this is still correct and will not need to be changed (to catch crazy cases) if we move to dynamic bitsets with lengths that are not the full width of a type.